src/pages/write-smart-contracts/setting-up-local-environment.md

```diff
@@ -0,0 +1,206 @@
+description: Learn how to set up your local smart contract development environment
+duration: 30 minutes
+experience: intermediate
+---
+
+## Introduction
+
+As soon as you get to work on your smart contracts and testing them end to end within your application, you need a local development environment that supports all steps and helps you achieve faster iteration cycles. This is critical because transaction times on the testnet take a few minutes and would slow you down. The solution is to run a local version of the testnet, often called a "mocknet".
+
+By the end of this tutorial, you will:
+
+- Have all required tooling and services up and running locally
+- Be able to deploy and test smart contracts inside your app
+
+## Prerequisites
+
+### Prior tutorials
+
+You need to be familiar with Clarity and the Stacks authentication (using Connect). Please finish all previous tutorials, especially the [Counter tutorial](/smart-contracts/counter-tutorial) and the [Building a todo app tutorial](/authentication/building-todo-app).
+
+### Environment
+
+You should have [NodeJS](https://nodejs.org/en/download/) and [Docker](https://docs.docker.com/get-docker/) installed on your workstation.
+
+## Overview
+
+For a complete local mocknet setup, you need the following services:
+
+- [Stacks Node](https://github.com/blockstack/stacks-blockchain)
+- [Stacks Blockchain API](https://github.com/blockstack/stacks-blockchain-api)
+- [Stacks Explorer](https://github.com/blockstack/explorer)
+
+## Step 1: Spin up mocknet
+
+The easiest way to spin up the Node, Blockchain API, and the Explorer locally is with Docker:
+
+```bash
+git clone https://github.com/blockstack/stacks-local-dev
+cd stacks-local-dev/
+docker-compose up -d
+```
+
+-> Make sure your Docker service is up and running
+
+The initial start will take a few minutes. As soon as it is complete, you should be able to open up [http://localhost:3999/](http://localhost:3999/) and see:
+
+```js
+{"status":"ready"}
+```
+
+~> The Stacks Node is running on port `20443`, the Stacks Blockchain API on `3999` and the Stacks Explorer is on port `3000`.
+
+## Step 2: Deploy your contract
+
+~> Make sure you completed the [Counter tutorial](/smart-contracts/counter-tutorial). A working contract and test-driven development setup is recommended
+
+When you develop your contract in a test-driven fashion, you will aim to complete the contract before deploying it to the testnet. Only once the contract is ready to be tested inside your app (or together with other contracts), you will need to deploy it. The fastest way to do so is by using the CLI:
+
+```shell
+# install the new CLI if not done yet
+npm install --global @stacks/cli
+
+# deploy the contract:
+# deploy_contract <SOURCE_FILE> <CONTRACT_NAME> <FEE> <NONCE> <PAYMENT_KEY> -t -T <NETWORK_ADDRESS>
+stx deploy_contract ./counter.clar counter 2000 0 cb3df38053d132895220b9ce471f6b676db5b9bf0b4adefb55f2118ece2478df01 -t -T http://localhost:20443
```

Comment:
@CharlieC3 @wileyj I tried the explorer sandbox and the CLI to deploy to the mocknet but I never get any transaction 🤔 is this image maybe outdated/broken? are you able to broadcast a tx to a mocknet?